src/Stan/Report/Html.hs

```diff
@@ -0,0 +1,68 @@
+{-# OPTIONS_GHC -Wno-missing-signatures #-}
+
+{-# LANGUAGE NoOverloadedStrings #-}
+
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+HTML to be generated in the report.
+-}
+
+module Stan.Report.Html
+    ( stanHtml
+    ) where
+
+import Clay (render)
+import Data.Char (toLower)
+import Html (body_, div_A, doctype_, footer_, h1_, h2_, head_, header_, html_, main_, meta_A, p_,
+             pre_, style_, title_, ( # ))
+
+import Stan.Report.Css (stanCss)
+
+import qualified Data.List as List (words)
+import qualified Html.Attribute as A
+
+
+stanHtml an = doctype_ # html_ (stanHead # stanBody)
+  where
+    stanBody = body_ (stanHeader # stanMain an # stanFooter)
+
+stanHeader = header_ (h1_ "Stan Report")
+
+stanMain an = main_
+    ( divIdClass "general-info" ""
+      ( divIdClassH "Project Info" "" ()
+      # divIdClassH "Stan Info" "" ()
+      # divIdClassH "Analysis Info" "" ()
+      )
+    # divIdClassH "Graphs" "" (p_ "Maybe later")
+    # divIdClassH "Observations" "" (pre_ an)
+    # divIdClassH "Configurations" "" (p_ "Later")
+    # divIdClassH "Summary" "" (p_ "Later")
+    # divIdClassH "Inspections" "" (p_ "Later")
+    )
+
+divIdClass i c = div_A (A.id_ i # A.class_ c)
```

Comment:
This particular helper function looks small. Just curious, what is the type of this function will be? Is it horrible even for such small and innocent looking helpers?

Reply:
The type of the function is:

```haskell
divIdClass :: String -> String -> b -> (Div :@: ('IdA := String # 'ClassA := String)) b
```

I don't see any value in it, so I decided to omit, as it only makes the maintenance the nightmare now (as I change everything so much).. However, let me know if it would make sense to specify it explicitly.

Reply:
@vrom911 The type doesn't look too useful... The type signature is much bigger than the implementation, and it looks like some redundant boilerplate... I agree that it's better to omit types in this case. HTML doesn't have types either, so we're not doing worse without types, but hopefully gain some extra safety from the library and some reusability by being able to define functions.